nixos/lib/make-disk-image.nix

```diff
@@ -402,11 +402,13 @@ let format' = format; in let
         done
       else
         mkdir -p $root/$(dirname $target)
-        if ! [ -e $root/$target ]; then
-          rsync $rsync_flags $source $root/$target
-        else
+        if [ -e $root/$target ]; then
           echo "duplicate entry $target -> $source"
           exit 1
+        elif [ -d $source ]; then
+          rsync $rsync_flags $source/ $root/$target
```

Comment:
Please add a comment to explain the rsync behavioral differences (everyone always gets tripped on this.)

Reply:
Done.